library/src/main/java/com/getstream/sdk/chat/view/messageinput/AttachmentsController.kt

```diff
@@ -0,0 +1,301 @@
+package com.getstream.sdk.chat.view.messageinput
+
+import android.net.Uri
+import androidx.annotation.VisibleForTesting
+import com.getstream.sdk.chat.R
+import com.getstream.sdk.chat.adapter.FileAttachmentListAdapter
+import com.getstream.sdk.chat.adapter.FileAttachmentSelectedAdapter
+import com.getstream.sdk.chat.adapter.MediaAttachmentAdapter
+import com.getstream.sdk.chat.adapter.MediaAttachmentSelectedAdapter
+import com.getstream.sdk.chat.enums.MessageInputType
+import com.getstream.sdk.chat.infrastructure.DispatchersProvider
+import com.getstream.sdk.chat.model.AttachmentMetaData
+import com.getstream.sdk.chat.model.ModelType
+import com.getstream.sdk.chat.utils.Constant
+import com.getstream.sdk.chat.utils.PermissionChecker
+import com.getstream.sdk.chat.utils.StorageHelper
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+internal class AttachmentsController(
+    private val rootController: MessageInputController,
+    private val permissionChecker: PermissionChecker,
+    private val storageHelper: StorageHelper,
+    private val dispatchersProvider: DispatchersProvider,
+    private val view: MessageInputView,
+    private val totalMediaAttachmentAdapter: MediaAttachmentAdapter,
+    private val totalFileAttachmentAdapter: FileAttachmentListAdapter,
+    private val selectedMediaAttachmentAdapter: MediaAttachmentSelectedAdapter,
+    private val selectedFileAttachmentAdapter: FileAttachmentSelectedAdapter,
+    private val showOpenAttachmentsMenuConfig: Boolean
+) {
+    internal var selectedAttachments: Set<AttachmentMetaData> = emptySet()
+        private set
```

Comment:
what is that?

Reply:
I expose this field for reading but protect for writing. Outside it's visible only as an immutable field (live `val`)